### PR TITLE
Menu Items for Disabling Rendering and Physics

### PR DESCRIPTION
**Pull request by luisbermudez**
(1) Added a new menu option that when checked disables graphics rendering such
that the image on the screen is not redrawn until the menu option is disabled.

(2) Added a new menu option that when checked disables the physics collisions such 
that the avatar does not collide into collidable objects until the menu option is disabled.
Demo Video: https://youtu.be/fGQT_qF2YBc

(3) Added a new menu option that when checked disables the physics simulation
such that moving objects stop until the menu option is disabled.
Demo Video: https://youtu.be/BSq8LaDcECg

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by ZappoMan** [comment]
ok to test

**Comment by luisbermudez** [comment]
Hello, Thank you. I have tested the interface application and it works fine. However, it seems to sometimes cache the collision objects once created -- regarding (2) above. Also, if there is a better or preferred code section to check menu items values (different from Application::paintGL), feel free to let me know. Any relevant feedback is highly appreciated.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ca848e71128864c18da1320564616f6e8dbc6b60 available at: https://gist.github.com/4f03544d14d5b0798004<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.dmg

File: libraries/physics/src/ToggleCollisionDispatcher.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#ifndef hifi_ToggleCollisionDispatcher_h
+#define hifi_ToggleCollisionDispatcher_h
+
+#include <btBulletDynamicsCommon.h>
+#include <BulletCollision/CollisionDispatch/btGhostObject.h>
+
+#include "Menu.h"
```
Comment (ZappoMan) [inline libraries/physics/src/ToggleCollisionDispatcher.h]: I'm not sure you need this... and it's causing a compiler error on windows....

```
113>d:\jenkins\workspace\pull-request-win\libraries\physics\src\ToggleCollisionDispatcher.h(17): fatal error C1083: Cannot open include file: 'Menu.h': No such file or directory (..\..\..\libraries\physics\src\ObjectMotionState.cpp) [D:\jenkins\workspace\pull-request-win\build\libraries\physics\physics.vcxproj]
   113>d:\jenkins\workspace\pull-request-win\libraries\physics\src\ToggleCollisionDispatcher.h(17): fatal error C1083: Cannot open include file: 'Menu.h': No such file or directory (..\..\..\libraries\physics\src\PhysicsEngine.cpp) [D:\jenkins\workspace\pull-request-win\build\libraries\physics\physics.vcxproj]
   113>d:\jenkins\workspace\pull-request-win\libraries\physics\src\ToggleCollisionDispatcher.h(17): fatal error C1083: Cannot open include file: 'Menu.h': No such file or directory (..\..\..\libraries\physics\src\PhysicalEntitySimulation.cpp) [D:\jenkins\workspace\pull-request-win\build\libraries\physics\physics.vcxproj]
   113>d:\jenkins\workspace\pull-request-win\libraries\physics\src\ToggleCollisionDispatcher.h(17): fatal error C1083: Cannot open include file: 'Menu.h': No such file or directory 
```

**Comment by ZappoMan** [comment]
@luisbermudez - my comment "ok to test" was just a message to tell our automated build system to do a build of your code.

**Comment by luisbermudez** [comment]
Excellent! Thank you for finding that error. I'm building on a Mac, so Xcode didn't catch it. I will remove Line 17 as a fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ca848e71128864c18da1320564616f6e8dbc6b60 available at: https://gist.github.com/4f03544d14d5b0798004<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.dmg

**Comment by luisbermudez** [comment]
The latest commit should have some unnecessary include statements removed. That should fix the Windows errors.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2fbed4c96712a4188724ef50fbfb119dce101e68 available at: https://gist.github.com/4f03544d14d5b0798004<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2fbed4c96712a4188724ef50fbfb119dce101e68 available at: https://gist.github.com/4f03544d14d5b0798004<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.dmg

File: libraries/physics/src/PhysicsEngine.cpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ PhysicsEngine::~PhysicsEngine() {
 void PhysicsEngine::init() {
     if (!_dynamicsWorld) {
         _collisionConfig = new btDefaultCollisionConfiguration();
-        _collisionDispatcher = new btCollisionDispatcher(_collisionConfig);
+        _collisionDispatcher = new ToggleCollisionDispatcher(_collisionConfig);
```
Comment (AndrewMeadows) [inline libraries/physics/src/PhysicsEngine.cpp]: There is a link to the automatic build logs on the conversation page.  According to the logs there is a compile error (for Xcode build) here at PhysicsEngine::56.

**Comment by luisbermudez** [comment]
Thanks, I see now. The build console output has Build Errors:
"cannot convert argument 1 from 'btDefaultCollisionConfiguration *' to 'const ToggleCollisionDispatcher &'"
I'll change it to an explicit constructor declaration in ToggleCollisionDispatcher class. Then, I'll check the automatic build logs for errors.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 70a3c529d8ead0f3aff4c43f5fdfda275fe6af9a available at: https://gist.github.com/4f03544d14d5b0798004<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6573.dmg

**Comment by luisbermudez** [comment]
It looks like all checks have passed this time around. 
Let me know if there's anything else I might have missed.

**Comment by AndrewMeadows** [comment]
The code compiles and looks good.  I'm closing now and will test.

**Pull request closed without merging**